tests/integration/test_charm.py

```diff
@@ -47,3 +53,23 @@ async def test_waits_for_config_server(ops_test: OpsTest) -> None:
         status="Missing relation to config-server.",
         timeout=300,
     )
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_mongos_external_connections(ops_test: OpsTest) -> None:
+    """Tests that mongos is accessible externally."""
+    for unit_id in range(len(ops_test.model.applications[MONGOS_APP_NAME].units)):
+        assert_node_port_available(
+            ops_test, node_port_name=f"{MONGOS_APP_NAME}-{unit_id}-external"
+        )
+
+        # TODO add this in once DPE-5040 / PR #20 merges
+        # exposed_node_port = get_port_from_node_port(ops_test, node_port_name="mongos-k8s-nodeport")
+        # public_k8s_ip = get_public_k8s_ip()
+        # username, password = await get_mongos_user_password(ops_test, MONGOS_APP_NAME)
+        # external_mongos_client = MongoClient(
+        #     f"mongodb://{username}:{password}@{public_k8s_ip}:{exposed_node_port}"
+        # )
+        # external_mongos_client.admin.command("usersInfo")
+        # external_mongos_client.close()
```

Comment:
**suggestion:** I know you will expand on this test later on, but something you can do now to verify that the node-port is actually reachable, would be to run some command like:
```bash
nc -zv {node-ip} {node-port}
```
Can run that as a subprocess.check_output, 0 exit would be a successful reach, 1 exit would be a failure.

Reply:
Thanks for this advice! Given that this check is pending PR #20 , which should be merged by tomorrow. I am going to skip it for now. But I appreciate the advice nonetheless

But have no fear I ran this test earlier today based on PR #20 and it worked great 😉